ocf_data_sampler/torch_datasets/datasets/pvnet_uk.py

```diff
@@ -47,12 +49,14 @@ def process_and_combine_datasets(
     numpy_modalities = []
 
     if "nwp" in dataset_dict:
-
         nwp_numpy_modalities = dict()
 
         for nwp_key, da_nwp in dataset_dict["nwp"].items():
-            # Standardise
+            # Validate channels before standardisation
             provider = config.input_data.nwp[nwp_key].provider
+            validate_channels(da_nwp, config, "nwp", provider)
```

Comment:
I might have mis understood, but weren't we going to validate that `NWP_MEANS` has the correct channels in it. In the bug with `sde` / `sd` it did not and we lost and channel without knowing. 

So we could have something like 
```
assert set(da_nwp .channels) in set(NWP_MEANS[provider].keys())
```
(I havent tested this out, this is just a guess, but hopefully it helps)

Reply:
For checking the channels are in the data, we should do that when the channels are selected, this would be in [load](https://github.com/openclimatefix/ocf-data-sampler/blob/main/ocf_data_sampler/load/load_dataset.py#L32), but this might already cover it